jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1191,10 +1191,10 @@ YUI.add('juju-view-utils', function(Y) {
     @param {Boolean} isBundle Whether or not this is an icon for a bundle.
   */
   utils.getIconPath = function(charmId, isBundle, env) {
-    var cfg = window.juju_config,
-        charmstoreURL = (cfg && cfg.charmstoreURL) || '',
-        localIndex = charmId.indexOf('local:'),
+    var charmstoreURL = this.get('charmstoreURL');
```

Comment:
This won't work because we aren't in the context of the app instance. you'll either want to do something like this:

```JavaScript
// If in app.js
utils.getIconPath.call(this, charmId, isBundle, env);

// If passing to another instance from app.js
utils.getIconPath.bind(this);

// Or you can require charmstoreURL as an argument when calling getIconPath.
utils.getIconPath. bind(this, this.get('charmstoreURL'));
```



```